src/axom/inlet/examples/nested_structs.cpp

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/inlet.hpp"
+
+#include <iostream>
+
+#include "CLI11/CLI11.hpp"
+#include "fmt/fmt.hpp"
+
+#include "axom/slic/core/SimpleLogger.hpp"
+
+namespace inlet = axom::inlet;
+using Vector = inlet::FunctionType::Vec3D;
+
+// Used to convert an unordered map (the type Inlet uses) to represent arrays
+// to a geometric vector type
+// FIXME: Clean this up/remove when PR #429 is merged
+Vector mapToVector(const std::unordered_map<int, double>& map)
+{
+  Vector result;
+  // Make sure the elements are accessed in order
+  for(int i = 0; i < 3; i++)
+  {
+    const auto ele = map.find(i);
+    if(ele != map.end())
+    {
+      result[i] = ele->second;
+    }
+  }
+  return result;
+}
+
+// A union of the members required for each of the operations is stored for simplicity
+struct Operator
+{
+  double rotate;
+  Vector axis;
+  Vector center;
+  Vector translate;
+  double x;
+  double y;
+  double z;
+  enum class Type
+  {
+    Translate,
+    Rotate,
+    Slice
+  } type;
+
+  // Again, the union of the necessary members are defined as part of the schema
+  static void defineSchema(inlet::Table& table)
+  {
+    // The rotation is in degrees
+    table.addDouble("rotate").range(-180, 180);
+    // Vectors are defined as arrays of doubles
+    table.addDoubleArray("axis");
+    table.addDoubleArray("center");
+    table.addDoubleArray("translate");
+    // The slice operation can have sub-entries, so we represent it as a struct
+    // Note that Inlet does not require a 1-1 correspondence (or any correspondence)
```

Comment:
The note here about not requiring a 1-1 correspondence is useful. Thanks for including it.